### ISSUE TITLE
lxc-create: Say image not found when not found in registry

### ISSUE THREAD

**Issue by rakotomandimby** [lxc/lxc#3878]
# Required information

 * Distribution: any
 * `lxc-start --version` : `4.0.9`

# Issue description

When trying to `lxc-create` a too old version that is not in the registry anymore, it gives a GPG error instead of saying that the image doe not exist in the registry:
```
# lxc-create -n d9-simserver -t download -- -d debian -r stretch --arch amd64
Setting up the GPG keyring
ERROR: Unable to fetch GPG key from keyserver
lxc-create: d9-simserver: lxccontainer.c: create_run_template: 1621 Failed to create container from template
lxc-create: d9-simserver: tools/lxc_create.c: main: 319 Failed to create container d9-simserver
```

**Comment by brauner** [comment]
Are you sure? The code doesn't look like this is possible and:
```
> lxc-create woody -t download -l trace -- -a amd64 -d debian -r woody
Setting up the GPG keyring
Downloading the image index
ERROR: Couldn't find a matching image
lxc-create: woody: lxccontainer.c: create_run_template: 1616 Failed to create container from template
lxc-create: woody: tools/lxc_create.c: main: 319 Failed to create container woody
```
seems to indicate this works correctly. My guess is that this is fixed by https://github.com/lxc/lxc/pull/3882 .

**Comment by bayuah** [comment]
Probably you can change the keyserver by yourself if your repository still use old package.

**Method 1:** Change the template file.
The template located in `/usr/share/lxc/templates/lxc-download`
Find `DOWNLOAD_KEYSERVE` variable.

```
if [ -z "${DOWNLOAD_KEYSERVER:-}" ]; then
  DOWNLOAD_KEYSERVER="hkp://pool.sks-keyservers.net"

  # Deal with GPG over http proxy
  if [ -n "${http_proxy:-}" ]; then
    DOWNLOAD_KEYSERVER="hkp://p80.pool.sks-keyservers.net:80"
    DOWNLOAD_GPG_PROXY="--keyserver-options http-proxy=\"${http_proxy}\""
  fi
fi

```

**Method 2:** Use environment.
You can use environment way to do it.
```
$ export DOWNLOAD_KEYSERVER="keyserver.ubuntu.com"
$ lxc-create -t download -n containername -- -d ubuntu -r focal -a amd64
```
or
```
$ DOWNLOAD_KEYSERVER="keyserver.ubuntu.com"  lxc-create
```

**Method 3:** Use argument.
Use `--keyserver` command line argument.
```
$ lxc-create -t download -n containername -- -d ubuntu -r focal -a amd64 --keyserver keyserver.ubuntu.com
```
I hope it will help someone, that probably still use old package.

By the way, the keyserver is already fixed in #3882.
